feature_PRESIDECMS-915 ::

cfmitrah opened the pull request:
- Remove unwanted \n \t from json ajaxproxy results.

cfmitrah committed:
feature_PRESIDECMS-915 ::

   - Remove unwanted \n \t from json ajaxproxy results.

DominicWatson commented on system/services/helpers/JQueryDatatablesHelpers.cfc:
```diff
@@ -29,6 +29,9 @@ component {
 			ArrayAppend( result.aaData, record );
 		}
 
+		if( isStruct( result.aaData[1] ) AND result.aaData[1]._options NEQ '' )
```
I'm not sure about this. Firstly:

* This presumably should be run against each row
* The cleanup needs to be a little smarter than just removing all whitespace?

I imagine the whitespace cleanup would be something like:

```reReplace( result.aaData[ i ]._options, "\s+", " ", "all" )

(i.e. tidyup double spaces, not get rid of all).

Would be good to see some before and after output after making this fix.

DominicWatson reviewed (changes requested):
Hi guys, thanks for this - could you take a look at the review feedback?

cfmitrah committed:
feature_PRESIDECMS-915 ::

- Functionality optimized.

cfmitrah commented:
- Functionality optimized. ( now it's  run against each row )
- yes that functionality removing all white space.
- I've attached screen samples for before and after changes.
**Before change :** 
![before](https://user-images.githubusercontent.com/532401/41218028-5fe2745c-6d77-11e8-9b6b-9a8be0b8602e.JPG)

**After change :**

![after](https://user-images.githubusercontent.com/532401/41218055-7578b772-6d77-11e8-9c8c-14971d5bd510.JPG)

Please check that and let me know if any update needed ?

DominicWatson commented:
The issue with removing _all_ whitespace is that text gets broken. There can be editorial text in the return (as in your example) and all the spaces are removed from that.

So, yes - needs to not remove _all_ whitespace, but rather convert 1 or more consecutive whitespace characters into a single whitespace character.

cfmitrah committed:
feature_PRESIDECMS-915 ::

- modify functionality to convert 1 or more consecutive whitespace characters into a single whitespace character.

DominicWatson commented:
Garrrr, this got auto closed because the target branch has disappeared. Really sorry for this. You should be able to re-open from your branch though by targeting the `stable` branch.
